src/csharp/Grpc.Core.Api/IAsyncStreamWriter.cs

```diff
@@ -36,6 +37,18 @@ public interface IAsyncStreamWriter<in T>
         /// <param name="message">The message to be written. Cannot be null.</param>
         Task WriteAsync(T message);
 
+#if NETSTANDARD2_1_OR_GREATER
+        /// <summary>
+        /// Writes a message asynchronously. Only one write can be pending at a time.
+        /// </summary>
+        /// <param name="message">The message to be written. Cannot be null.</param>
+        /// <param name="cancellationToken">Cancellation token that can be used to cancel the operation.</param>
+        Task WriteAsync(T message, CancellationToken cancellationToken)
```

Comment:
Kestrel supports aborting in-progress requests. I haven't looked into exactly how it will work but it will either be:
1. The cancellation token is passed to the underlying writer and if the writer gets a cancellation then it will abort the request, or
2. WriteAsync subscribes to the cancellation token with a callback that calls `HttpContext.Abort()`.

I don't know the internals of Grpc.Core.

Reply:
> Grpc.Core

Would Grpc.Core cancellation be supported with this code:

https://github.com/grpc/grpc/blob/84101427d0a699e65837af43f4f0ac326c7a2aca/src/csharp/Grpc.Core/Internal/AsyncCallBase.cs#L393-L399

It is used by `IAsyncStreamReader<T>.MoveNext` which takes a cancellation token:

https://github.com/grpc/grpc/blob/2d4f3c56001cd1e1f85734b2f7c5ce5f2797c38a/src/csharp/Grpc.Core/Internal/ServerRequestStream.cs#L50-L58

Reply:
> Grpc.Core

Actually, Grpc.Core only supports up to netstandard20 and net45. The new overload won't be available.